source/common/protobuf/utility.cc

```diff
@@ -39,6 +39,79 @@ void blockFormat(YAML::Node node) {
   }
 }
 
+ProtobufWkt::Value parseYamlNode(const YAML::Node& node) {
+  ProtobufWkt::Value value;
+  switch (node.Type()) {
+  case YAML::NodeType::Null:
+    value.set_null_value(ProtobufWkt::NULL_VALUE);
+    break;
+  case YAML::NodeType::Scalar: {
+    if (node.Tag() == "!") {
+      value.set_string_value(node.as<std::string>());
+      break;
+    }
+    bool bool_value;
+    if (YAML::convert<bool>::decode(node, bool_value)) {
+      value.set_bool_value(bool_value);
+      break;
+    }
+    int64_t int_value;
+    if (YAML::convert<int64_t>::decode(node, int_value)) {
+      if (std::numeric_limits<int32_t>::min() <= int_value &&
+          std::numeric_limits<int32_t>::max() >= int_value) {
+        // We could convert all integer values to string but it will break some stuff relying on
+        // ProtobufWkt::Struct itself, only convert small numbers into number_value here.
```

Comment:
Not sure I fully grok this..

Reply:
E.g. `{"a": "0"}` and `{"a": 0}` result different `ProtobufWkt::Struct` in the `oneof` (`string_value` or `number_value`), but if you parse it into message:
```
message Foo {
  int64 a = 1;
}
```
then it makes no difference.

There are some places expect the former so I implemented like this. We can kill this and always encode them string (and fix tests), but seems too much in this PR.

Reply:
Fair enough, thanks for the explanation.